servers/src/main/java/tachyon/master/Master.java

```diff
@@ -74,4 +75,13 @@
    * @throws IOException
    */
   void stop() throws IOException;
+
+  /**
+   * Provides the master with a {@link Journal} capable of writing, in preparation to starting as
+   * the leader. This enables transitioning from standby master to leader master without having to
+   * construct a new master object.
+   *
+   * @param journal the {@link Journal} capable of writing
+   */
+  void upgradeToWriteJournal(Journal journal);
```

Comment:
This reads a little weird, given the hierarchy of journal. Right now the readonly journal extends read-write journal, we may consider the opposite: let read-write journal extend the readonly version.